src/main/scala/chisel3/aop/Select.scala

```diff
@@ -43,7 +43,13 @@ object Select {
           else
             Seq.empty
         head ++ collect(ifRegion)(f) ++ collect(elseRegion)(f)
-      case LayerBlock(_, _, region)  => collect(region)(f)
+      case cmd @ LayerBlock(_, _, region) =>
+        val head =
+          if (f.isDefinedAt(cmd))
+            Seq(f(cmd))
+          else
+            Seq.empty
+        head ++ collect(region)(f)
```

Comment:
```suggestion
        val head = f.lift(cmd).toSeq
        head ++ collect(region)(f)
```
You can `lift` a `PartialFunction` to return `Option[R]`, then just `.toSeq` to get a `Seq`. You could alternatively leave it as an `Option` and prepend with `head ++: collect(region)(f)` but either way is fine.

Reply:
Awesome, thanks! I reached for that but wasn't entirely sure and so just copied this from above.
I like your suggestion, thanks!